lib/actions/device.coffee

```diff
@@ -266,27 +275,82 @@ exports.init =
 
 			(confirmed, callback) ->
 				return done() if not confirmed
-				options.yes = confirmed
+				return callback() if networkOptions.network?
+				visuals.patterns.selectNetworkParameters (error, parameters) ->
+					return callback(error) if error?
+					_.extend(networkOptions, parameters)
+					return callback()
+
+			(callback) ->
+				console.info("Checking application: #{options.application}")
+				resin.models.application.get(options.application, callback)
+
+			(application, callback) ->
+				async.parallel
+
+					manifest: (callback) ->
+						console.info('Getting device manifest for the application')
+						resin.models.device.getManifestBySlug(application.device_type, callback)
+
+					config: (callback) ->
+						console.info('Fetching application configuration')
+						resin.models.application.getConfiguration(options.application, networkOptions, callback)
 
-				tmp.file
-					prefix: 'resin-image-'
-					postfix: '.img'
 				, callback
 
-			(tmpPath, tmpFd, cleanupCallback, callback) ->
-				options.output = tmpPath
+			(results, callback) ->
+				console.info('Associating the device')
 
-				# TODO: Figure out how to make use of capitano.run()
-				# here given the complexity of converting network
-				# params object to string options
-				osAction.download.action params, options, (error, outputFile) ->
+				registerDevice.register pine, results.config, (error, device) ->
```

Comment:
This can be:
```coffee
registerDevice.register(pine, results.config)
.then (device) ->
	results.config.deviceId = device.id
	params.uuid = results.config.uuid = device.uuid
.return result
.nodeify(callback)
```